src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -34,11 +33,24 @@ internal static SafeSslHandle AllocateSslContext(long options, SafeX509Handle ce
 
                 libssl.SSL_CTX_set_quiet_shutdown(innerContext, 1);
 
+                libssl.SSL_CTX_set_cipher_list(innerContext, encryptionPolicy);
+
                 if (certHandle != null && certKeyHandle != null)
                 {
                     SetSslCertificate(innerContext, certHandle, certKeyHandle);
                 }
 
+                if (remoteCertRequired)
+                {
+                    Debug.Assert(isServer, "isServer flag should be true");
+                    libssl.SSL_CTX_set_verify(innerContext,
+                    (int)libssl.ClientCertOption.SSL_VERIFY_PEER |
+                    (int)libssl.ClientCertOption.SSL_VERIFY_FAIL_IF_NO_PEER_CERT, verify_callback);
```

Comment:
Yes, the delegate must be kept alive.  The naming is confusing here, but it looks like verify_callback is a method, so the compiler's going to generate a temporary delegate.  You should manually cache this delegate into a static and then just use that field here.